File: client/types/student.ts
export type RiskLevel = "low" | "medium" | "high"

export interface RiskDTO {
  level: RiskLevel
  reasons: string[]
}

export interface RecordDTO {
  term: number
  attendance_pct: number
  grade_avg: number
  participation: number
}

export interface StudentListItem {
  id: string
  name: string
  school_name: string
  zone: "rural" | "urban"
  education_level: "primary" | "secondary"
  grade: number
  native_language: string
  risk: RiskDTO
}

export interface DistrictDTO {
  name: string
  department: string
  primary_dropout_rate: number
  secondary_dropout_rate: number
}

export interface StudentDetail extends StudentListItem {
  district: DistrictDTO
  records: RecordDTO[]
}

export interface AssessResponse {
  id: string
  student_id: string
  risk_level: RiskLevel
  reasons: string[]
  assessed_at: string
}

File: server/internal/domain/student.go
package domain

import "time"

type RiskLevel string

const (
	RiskLow    RiskLevel = "low"
	RiskMedium RiskLevel = "medium"
	RiskHigh   RiskLevel = "high"
)

type RiskResult struct {
	Level   RiskLevel
	Reasons []string
}

type AcademicRecord struct {
	Term          int
	AttendancePct float64
	GradeAvg      float64
	Participation int
}

type StudentSummary struct {
	ID             string
	Name           string
	SchoolName     string
	Zone           string
	EducationLevel string
	Grade          int
	NativeLanguage string
	Records        []AcademicRecord
}

type StudentDetail struct {
	StudentSummary
	Department           string
	DistrictName         string
	PrimaryDropoutRate   float64
	SecondaryDropoutRate float64
}

type RiskAssessment struct {
	ID         string
	StudentID  string
	Level      RiskLevel
	Reasons    []string
	AssessedAt time.Time
}
